packages/block-library/src/heading/block.json

```diff
@@ -30,49 +30,15 @@
 		},
 		"fontSize": true,
 		"lineHeight": true,
-		"__experimentalSelector": {
-			"core/heading/h1": {
-				"selector": "h1",
-				"title": "h1",
-				"attributes": {
-					"level": 1
-				}
-			},
-			"core/heading/h2": {
-				"selector": "h2",
-				"title": "h2",
-				"attributes": {
-					"level": 2
-				}
-			},
-			"core/heading/h3": {
-				"selector": "h3",
-				"title": "h3",
-				"attributes": {
-					"level": 3
-				}
-			},
-			"core/heading/h4": {
-				"selector": "h4",
-				"title": "h4",
-				"attributes": {
-					"level": 4
-				}
-			},
-			"core/heading/h5": {
-				"selector": "h5",
-				"title": "h5",
-				"attributes": {
-					"level": 5
-				}
-			},
-			"core/heading/h6": {
-				"selector": "h6",
-				"title": "h6",
-				"attributes": {
-					"level": 6
-				}
-			}
+		"__experimentalSelector": "h1,h2,h3,h4,h5,h6",
+		"__experimentalSelectorsForElements": {
```

Comment:
Why do we need this?

Reply:
I've updated the PR description with more context. See "How blocks can re-define the element's selectors". The idea is that the default behavior of elements (append to the block selector) doesn't work well in a few blocks, so they need to re-define how these work.

Reply:
Yeah, I can see that. That said, I'm not sold on the idea yet and I do wonder whether we need to support it. The problem basically is when a block maps to the tagName directly.

h2, h3, h4 are  not "inside" the heading block so for me it doesn't make sense to map the elements config inside that block to the block wrapper itself. The **inside** there is important IMO. Now this does mean there are things that were possible before that are not possible right now. I kind of want to see whether the global  "elements" are enough for these use-cases. In other words do people want to target `heading h1` or they want just to target `h1` cross blocks.

Do you think we can consider starting without it?

Reply:
If we remove the ability for blocks to change the default behavior we have the following

things that will still be possible:

- themes can still style individual heading elements (h1...h6) via the top level elements
- they can also style "all headings" (`h1, h2, h3, h4, h5, h6`) at once via the `blocks.core/heading` (not possible with the previous format)
- they can also style "all versions" of `core/post-title` (`.wp-block-post-title`) and `core/query-title` (`.wp-block-query-title`) (not possible with the previous format)

things that won't work but worked with the previous format:

- have different styles for the different output that `core/post-title` and `core/query-block` can have ― I've only seen this in use for the `core/heading` block, for which we have a mechanism, so not sure if this is a _strong have_

things that won't work but we didn't have in the previous format either:

- link elements within `core/heading` (the CSS output won't make sense)

Judging by the use cases we won't cover, it seems it should be fine to ship this PR without this.

Reply:
> link elements within core/heading (the CSS output won't make sense)

I think we can fix this by splitting the selector over `,` and injecting the element selectors.

Reply:
Also @mtias you might have an opinion here.

Reply:
> I think we can fix this by splitting the selector over `,` and injecting the element selectors.

Makes sense for all elements, will do :+1:

Reply:
Done both things (split by block selector if is compounded + remove the ability for blocks to re-define element selectors). 

The migration from the previous format to the new one behaves this way:

- `core/heading/h1`...`core/heading/h6`
  - styles => converted to the corresponding `elements.h1`...`elements.h6` styles. Except for the link color, that will stay in `core/heading` if it's defined (as the elements can't nest other elements).
  - settings => converted to `blocks.core/heading` settings (later sub-selectors overwrites the previous: settings for `core/heading/h6` override settings for `core/heading/h5` and so on).
- `core/post-title/h1`...`core/post-title/h6` => for both styles and settings it's converted to `blocks.core/post-title` and later sub-selectors overwrite the previous.
- `core/query-title/h1`...`core/query-title/h6` => for both styles and settings it's converted to `blocks.core/query-title` and later sub-selectors overwrite the previous.